calico_containers/pycalico/datastore.py

```diff
@@ -113,6 +123,51 @@ def wrapped(*args, **kwargs):
     return wrapped
 
 
+def assert_valid_etcd_keys(scheme, key, cert, ca_cert):
+    """
+    Validate etcd SSL environment variables set on the host machine.  The key
+    and certificate values must both be set or not set.  All values passed
+    in must be readable.
+
+    Raise a DataStoreError if any assertions fail.
+
+    :param scheme: HTTP protocol type. One of: "", "http", or "https"
+    :param key: Path to secure key for etcd or None value
+    :param cert: Path to secure certificate for etcd or None value
+    :param ca_cert: Path to Certificate Authority signing certificate for etcd
+    of None value
+    """
+    if scheme == "https":
+        # etcd key and certificate must be both specified or both not specified
+        if bool(key) != bool(cert):
+            raise DataStoreError("Invalid %s, %s combination. Key and "
+                                 "certificate must both be specified or both "
+                                 "be blank. Values provided: %s=%s, %s=%s" %
+                                 (ETCD_KEY_FILE_ENV, ETCD_CERT_FILE_ENV,
+                                  ETCD_KEY_FILE_ENV, key,
+                                  ETCD_CERT_FILE_ENV, cert))
+        # Make sure etcd key and certificate are readable
+        if key and cert:
+            if not (os.access(key, os.R_OK) and os.access(cert, os.R_OK)):
+                raise DataStoreError("Cannot read %s and/or %s. Both must be "
+                                     "readable file paths. Values provided: "
+                                     "%s=%s, %s=%s" % (ETCD_KEY_FILE_ENV,
+                                                       ETCD_CERT_FILE_ENV,
+                                                       ETCD_KEY_FILE_ENV,
+                                                       key,
+                                                       ETCD_CERT_FILE_ENV,
+                                                       cert))
+            # If Certificate Authority cert file provided, check it's readable
+            if ca_cert and not os.access(ca_cert, os.R_OK):
+                raise DataStoreError("Cannot read %s. Value must be readable "
+                                     "file path. Value provided: %s" %
+                                     (ETCD_CA_CERT_FILE_ENV, ca_cert))
+    elif not (scheme == "http" or scheme == ""):
```

Comment:
I was wondering this as well.  I went back and forth with it a few times.  I think this is specific enough that it should only be used in this one case though, so checking this against "http" alone should be enough.